File: hw/aymacana/u3/hw33CrudStrategy/CrudStrategyPY/controller/validation_controller.py
import re

class ValidationController:
    @staticmethod
    def validate_name(name: str) -> bool:
        if not name:
            return False
        return bool(re.match("^[a-zA-ZáéíóúÁÉÍÓÚñÑ\\s]+$", name))
    
    @staticmethod
    def validate_apartment_number(apartment_number: str) -> bool:
        if not apartment_number:
            return False
        return bool(re.match("^[a-zA-Z0-9\\-]+$", apartment_number))
    
    @staticmethod
    def validate_email(email: str) -> bool:
        if not email:
            return False
        return bool(re.match("^[\\w.%+-]+@[\\w.-]+\\.[a-zA-Z]{2,}$", email))
    
    @staticmethod
    def validate_phone(phone: str) -> bool:
        if not phone:
            return False
        return bool(re.match("^\\d{7,15}$", phone))
    
    @staticmethod
    def validate_all_fields(name: str, apartment_number: str, email: str, phone: str) -> str:
        errors = []
        
        if not ValidationController.validate_name(name):
            errors.append("• Name can only contain letters and spaces.")
        
        if not ValidationController.validate_apartment_number(apartment_number):
            errors.append("• Apartment number can only contain letters, numbers, and hyphens.")
        
        if not ValidationController.validate_email(email):
            errors.append("• Email must have a valid format (example@domain.com).")
        
        if not ValidationController.validate_phone(phone):
            errors.append("• Phone can only contain numbers (7-15 digits).")
        
        return "\n".join(errors)

File: hw/toapanta/u3/hw35CrudStrategy/CrudStrategyJS/controllers/ValidationController.js
export class ValidationController {
    static validateName(name) {
        const nameRegex = /^[a-zA-ZáéíóúÁÉÍÓÚñÑ\s]+$/;
        return name && nameRegex.test(name);
    }

    static validateApartmentNumber(apartmentNumber) {
        const apartmentRegex = /^[a-zA-Z0-9\-]+$/;
        return apartmentNumber && apartmentRegex.test(apartmentNumber);
    }

    static validateEmail(email) {
        const emailRegex = /^[\w.%+-]+@[\w.-]+\.[a-zA-Z]{2,}$/;
        return email && emailRegex.test(email);
    }

    static validatePhone(phone) {
        const phoneRegex = /^\d{7,15}$/;
        return phone && phoneRegex.test(phone);
    }

    static validateAllFields(name, apartmentNumber, email, phone) {
        const errors = [];
        
        if (!this.validateName(name)) {
            errors.push("• Name can only contain letters and spaces.");
        }
        
        if (!this.validateApartmentNumber(apartmentNumber)) {
            errors.push("• Apartment number can only contain letters, numbers, and hyphens.");
        }
        
        if (!this.validateEmail(email)) {
            errors.push("• Email must have a valid format (example@domain.com).");
        }
        
        if (!this.validatePhone(phone)) {
            errors.push("• Phone can only contain numbers (7-15 digits).");
        }
        
        return errors.length > 0 ? errors.join("\n") : "";
    }
}